File: backend/src/main/java/com/company/Entities/Comment.java
package com.company.Entities;

public class Comment {
    Integer commentId, relatesToVideoId, likes, dislikes, responseToCommentId;
    String postedByUsername, content;

    public Integer getCommentId() {
        return commentId;
    }

    public void setCommentId(Integer commentId) {
        this.commentId = commentId;
    }

    public Integer getRelatesToVideoId() {
        return relatesToVideoId;
    }

    public void setRelatesToVideoId(Integer relatesToVideoId) {
        this.relatesToVideoId = relatesToVideoId;
    }

    public Integer getLikes() {
        return likes;
    }

    public void setLikes(Integer likes) {
        this.likes = likes;
    }

    public Integer getDislikes() {
        return dislikes;
    }

    public void setDislikes(Integer dislikes) {
        this.dislikes = dislikes;
    }

    public Integer getResponseToCommentId() {
        return responseToCommentId;
    }

    public void setResponseToCommentId(Integer responseToCommentId) {
        this.responseToCommentId = responseToCommentId;
    }

    public String getPostedByUsername() {
        return postedByUsername;
    }

    public void setPostedByUsername(String postedByUsername) {
        this.postedByUsername = postedByUsername;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    @Override
    public String toString() {
        return "Comment{" +
                "commentId=" + commentId +
                ", relatesToVideoId=" + relatesToVideoId +
                ", likes=" + likes +
                ", dislikes=" + dislikes +
                ", responseToCommentId=" + responseToCommentId +
                ", postedByUsername='" + postedByUsername + '\'' +
                ", content='" + content + '\'' +
                '}';
    }

    public Comment(Integer commentId, Integer relatesToVideoId, String postedByUsername, String content, Integer likes, Integer dislikes, Integer responseToCommentId){
        this.commentId = commentId;
        this.relatesToVideoId = relatesToVideoId;
        this.postedByUsername = postedByUsername;
        this.content = content;
        this.likes = likes;
        this.dislikes = dislikes;
        this.responseToCommentId = responseToCommentId;
    }
}

File: frontend/src/jsClasses/general/Comment.js
export default class Comments {
  constructor(
    commentId,
    relatesToVideoId,
    postedByUsername,
    content,
    likes,
    dislikes,
    respondedToCommentId,
    timeOfPosting
  ) {
    this.commentId = commentId;
    this.relatesToVideoId = relatesToVideoId;
    this.postedByUsername = postedByUsername;
    this.content = content;
    this.likes = likes;
    this.dislikes = dislikes;
    this.responseToCommentId = respondedToCommentId;
    this.timeOfPosting = timeOfPosting;
  }

  settimeOfPosting(timeOfPosting) {
    this.timeOfPosting = timeOfPosting;
  }

  gettimeOfPosting() {
    return this.timeOfPosting;
  }
  getRespondedToCommentId() {
    return this.respondedToCommentId;
  }

  setRespondedToCommentId(respondedToCommentId) {
    this.respondedToCommentId = respondedToCommentId;
  }

  getDislikes() {
    return this.dislikes;
  }

  setDislikes(dislikes) {
    this.dislikes = dislikes;
  }

  setLikes(likes) {
    this.likes = likes;
  }

  getLikes() {
    return this.likes;
  }

  setContent(content) {
    this.content = content;
  }

  getContent() {
    return this.content;
  }

  getPostedByUsername() {
    return this.postedByUsername;
  }

  setPostedByUsername(postedByUsername) {
    this.postedByUsername = postedByUsername;
  }

  getRelatesToVideoId() {
    return this.relatesToVideoId;
  }

  setRelatesToVideoId(videoId) {
    this.relatesToVideoId = this.relatesToVideoId;
  }

  getCommentId() {
    return this.commentId;
  }

  setCommentId(commentId) {
    this.commentId = commentId;
  }
}
